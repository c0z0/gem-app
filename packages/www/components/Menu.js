import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import Container from './Container'
import { Diamond } from './Svg'

const MenuContainer = styled(Container)`
  padding-top: 20px;
  margin-bottom: 12px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`

const Logo = styled.div`
  display: flex;
  align-items: center;
`

const Title = styled.span`
  margin-left: 8px;
  font-weight: 400;
  margin-right: 20px;
  color: ${({ theme }) => theme.main};
  font-size: 20px;
`

const StyledDiamond = styled(Diamond).attrs({
  alt: 'gem logo'
})`
  width: 40px;
`

const MenuItem = styled.a`
  color: #aaa;
  font-size: 14px;
  text-decoration: none;
  margin-left: 12px;
  transition: all 0.2s;

  &:hover {
    color: #484848;
  }
`

export default function Menu({ onLogout }) {
  return (
    <MenuContainer>
      <Logo>
        <StyledDiamond />
        <Title>Gem</Title>
      </Logo>
      <div>
        <MenuItem href="" onClick={onLogout}>
          Logout
        </MenuItem>
        <MenuItem href="https://github.com/c0z0/gem-app">[src]</MenuItem>
      </div>
    </MenuContainer>
  )
}

Menu.propTypes = {
  onLogout: PropTypes.func.isRequired
}
