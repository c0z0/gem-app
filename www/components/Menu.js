import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import Container from './Container'

const MenuContainer = styled(Container)`
  padding-top: 20px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`

const Logo = styled.div`
  display: flex;
  align-items: center;
`

const Title = styled.span`
  margin-left: 8px;
  font-weight: 400;
  margin-right: 20px;
  color: ${({theme}) => theme.main};
  font-size: 20px;
`

const Diamond = styled.img.attrs({
  src: '/static/diamond.svg',
})`
  width: 40px;
`

const MenuItem = styled.a`
  color: #aaa;
  font-size: 14px;
  text-decoration: none;
  margin-left: 12px;
  transition: all 0.2s;

  &:hover {
    color: #484848;
  }
`

export default function Menu({onLogout}) {
  return (
    <MenuContainer>
      <Logo>
        <Diamond />
        <Title>Gem</Title>
      </Logo>
      <div>
        <MenuItem href="" onClick={onLogout}>
          Logout
        </MenuItem>
        <MenuItem href="https://github.com/c0z0/gem-app">[src]</MenuItem>
      </div>
    </MenuContainer>
  )
}

Menu.propTypes = {
  onLogout: PropTypes.func.isRequired,
}
