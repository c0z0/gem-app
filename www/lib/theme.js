export default {
  main: '#75489B',
  mainGradient: 'linear-gradient(315deg,#75489b 0%,#a8708d 100%)',
  b: {
    phoneOnly: `max-width: 991px`,
    tabletUp: `min-width: 992px`,
  },
}
