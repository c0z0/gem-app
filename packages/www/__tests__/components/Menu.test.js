import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import theme from '../../lib/theme'
import 'jest-styled-components'

import Menu from '../../components/Menu'

afterEach(cleanup)

test('Menu matches snapshot', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Menu onLogout={() => {}} />
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})

test('Menu logs out', () => {
  const onLogout = jest.fn()

  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <Menu onLogout={onLogout} />
    </ThemeProvider>
  )

  fireEvent.click(getByText('Logout'))

  expect(onLogout).toHaveBeenCalled()
})
