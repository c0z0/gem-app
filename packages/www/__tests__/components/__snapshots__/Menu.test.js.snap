// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu matches snapshot 1`] = `
.c0 {
  max-width: 900px;
  margin: 0 auto;
  padding: 0 12px;
  padding-top: 20px;
  margin-bottom: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c3 {
  margin-left: 8px;
  font-weight: 400;
  margin-right: 20px;
  color: #75489B;
  font-size: 20px;
}

.c2 {
  width: 40px;
}

.c4 {
  color: #aaa;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-left: 12px;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c4:hover {
  color: #484848;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        alt="gem logo"
        class="c2"
        viewBox="0 0 512 512"
      >
        <path
          d="M360.129 172.138L256 472.276l256-300.138z"
          fill="#ffe182"
          stroke="#ffe182"
          stroke-width="1"
        />
        <g
          fill="#ffcd73"
          stroke="#ffe182"
          stroke-width="1"
        >
          <path
            d="M105.931 39.724L0 172.138h151.871zM360.129 172.138H512L406.069 39.724zM360.129 172.138L256 39.724 151.871 172.138z"
          />
        </g>
        <path
          d="M256 39.724H105.931l45.94 132.414z"
          fill="#ffaa64"
          stroke="#ffaa64"
          stroke-width="1"
        />
        <path
          d="M406.069 39.724H256l104.129 132.414z"
          fill="#ffe182"
          stroke="#ffe182"
          stroke-width="1"
        />
        <path
          d="M151.871 172.138L256 472.276l104.129-300.138z"
          fill="#ffaa64"
          stroke="#ffaa64"
          stroke-width="1"
        />
        <path
          d="M0 172.138l256 300.138-104.129-300.138z"
          fill="#ff8c5a"
          stroke="#ff8c5a"
          stroke-width="1"
        />
      </svg>
      <span
        class="c3"
      >
        Gem
      </span>
    </div>
    <div>
      <a
        class="c4"
        href="/portal"
      >
        Portal
      </a>
      <a
        class="c4"
        href=""
      >
        Logout
      </a>
      <a
        class="c4"
        href="https://github.com/c0z0/gem-app"
      >
        [src]
      </a>
    </div>
  </div>
</div>
`;
